tests/RobotFramework/tasks/E2E_Tutorials/connect_to_c8y_send_measurements.robot

```diff
@@ -0,0 +1,173 @@
+    Should Contain    ${output}    tedge-mapper-c8y service successfully started and enabled!
+    Should Contain    ${output}    Enabling software management.
+    Should Contain    ${output}    Checking if tedge-agent is installed.
+    Should Contain    ${output}    Starting tedge-agent service.
+    Should Contain    ${output}    Persisting tedge-agent on reboot.
+    Should Contain    ${output}    tedge-agent service successfully started and enabled!
+    Log    Connected to Cumulocity IoT and verified connection for ${DEVICE_SN}
+
+Send Temperature Measurement
+    [Documentation]    Send a temperature measurement.
+    Execute Command    tedge mqtt pub 'te/device/main///m/environment' '{"temperature": 25}'
+    Log    Sent temperature measurement: 25°C
+
+Send Three Phase Current Measurement
+    [Documentation]    Send a three-phase current measurement.
+    Execute Command    tedge mqtt pub 'te/device/main///m/electrical' '{"three_phase_current": {"L1": 9.5, "L2": 10.3, "L3": 8.8}}'
+    Log    Sent three-phase current measurement
+
+Send Combined Measurement
+    [Documentation]    Send a combined measurement.
+    Execute Command    tedge mqtt pub 'te/device/main///m/combined' '{"time": "2020-10-15T05:30:47+00:00", "temperature": 25, "current": {"L1": 9.5, "L2": 10.3, "L3": 8.8}, "pressure": 98}'
+    Log    Sent combined measurement
+
+Send Child Device Temperature Measurement
+    [Documentation]    Send a temperature measurement to a child device.
+    Execute Command    tedge mqtt pub 'te/device/child1///m/environment' '{"temperature": 25}'
+    Log    Sent temperature measurement to child device: 25°C
+
+Verify Measurement In Cumulocity
+    [Arguments]    ${type}    ${value}
+    [Documentation]    Verify the measurement in Cumulocity.
+    # Add implementation for checking the measurement in Cumulocity
+    Log    Verified ${type} measurement with value ${value} in Cumulocity
+
+Verify Combined Measurement In Cumulocity
+    [Documentation]    Verify the combined measurement in Cumulocity.
+    # Add implementation for checking the combined measurement in Cumulocity
+    Log    Verified combined measurement in Cumulocity
+
+Verify Child Device Measurement In Cumulocity
+    [Arguments]    ${child_id}    ${type}    ${value}
+    [Documentation]    Verify the child device measurement in Cumulocity.
+    # Add implementation for checking the child device measurement in Cumulocity
+    Log    Verified ${type} measurement for child device ${child_id} with value ${value} in Cumulocity
+
+Custom Setup
+    [Documentation]    Initializes the device environment. 
+    ...                Sets up the device, transfers necessary packages, 
+    ...                installs them, and configures Cumulocity for connectivity.
+    ${DEVICE_SN}=    Setup    skip_bootstrap=True
+    Set Suite Variable    ${DEVICE_SN}
+    ${log}    Transfer To Device    target/aarch64-unknown-linux-musl/packages/*.deb    /home/pi/
```

Comment:
This assumes you are running on an arm64/aarch64 system. We should be more dynamic and allow detecting the target architecture and then selecting the correct package to install.

Reply:
as stated in the other Task, I do use this exclusively with Raspberry

Reply:
Though the idea is to have something which can be applied to also different devices, and doesn't fail when using on a Raspberry Pi armv7 device...

Or alternatively, clearly state at the top of the task that this script is only intended to be run on Raspberry Pi devices with an aarch64/arm64 architecture.